### PR TITLE
adding draft of monte carlo chapter

### PR DESCRIPTION
**Pull request by leios**
This one took me a while to make the images. I had to find some old code and update it.

I couldn't decide between using a gif or three images side-by-side for showing that with more points, you get a better result. This version has 3 images. 

What do you guys think? For now, the book is online-only, so I think a gif is ultimately fine. There are a lot of other things we will need to change if we ever want to actually print this book... so maybe it's alright to have good visualizations?

### CODE REVIEW

File: chapters/monte_carlo/monte_carlo.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+So, if we can find the $$\text{Ratio}$$ and we know $$r$$, we should be able to easily find the $$\text{Area}_{\text{circle}}$$.
+The question is, "How do we easily find the $$\text{Ratio}$$?"
+Well, one way is with *random sampling*.
+We basically just pick a bunch of points randomly in the square, and
+each point is tested to see whether it's in the circle or not:
+
+{% method %}
+{% sample lang="jl" %}
+[import:2-8, lang:"julia"](code/julia/monte_carlo.jl)
+{% sample lang="cpp" %}
```
Comment (Gathros) [inline chapters/monte_carlo/monte_carlo.md]: Why have you got C++ here?

File: chapters/monte_carlo/monte_carlo.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+This saves a bit of computation time, but also requires us to multiply our output by $$4$$.
+
+That's all there is to it!
+Feel free to submit your version via pull request, and thanks for reading!
+
+{% method %}
+{% sample lang="jl" %}
+### Julia
+[import, lang:"julia"](code/julia/monte_carlo.jl)
+{% sample lang="cpp" %}
```
Comment (Gathros) [inline chapters/monte_carlo/monte_carlo.md]: And here?

**Comment by leios** [comment]
Both of those should have been removed. Thanks!

**Comment by leios** [comment]
Ok, typo fixed! Thanks for the catch!

**Comment by jiegillet** [comment]
How do you compile and visualize someone else's PR?

File: chapters/monte_carlo/monte_carlo.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+As long as you can write some function to tell whether the provided point is inside the shape you want (like `in_circle()` in this case), you can use monte carlo integration!
+This is obviously an incredibly powerful tool and has been used time and time again for many different areas of physics and engineering.
+I can gaurantee that we will see similar methods crop up all over the place in the future!
+
+# Example Code
+Monte carlo methods are famous for their simplicity.
+It doesn't take too many lines to get something simple going.
+Here, we are just integrating a circle, like we described above; however, there is a small twist and trick.
+Instead of calculating the area of the circle, we are instead trying to find the value of $$\pi$$, and
+rather than integrating the entire circle, we are only integrating the upper left quadrant of the circle from $$-1 < x,y < 1$$.
```
Comment (jiegillet) [inline chapters/monte_carlo/monte_carlo.md]: Do you mean upper right quadrant and do you mean $$0 < x,y < 1$$?

**Comment by Gathros** [comment]
@jiegillet go to the fork and the branch then clone it (on unix use this command `git clone -b <branch> <url>`) then `gitbook install` and then `gitbook serve`, then you can see the book at localhost:4000.

File: chapters/monte_carlo/monte_carlo.md (hunk 3 above)
Comment (leios) [reply chapters/monte_carlo/monte_carlo.md]: Typo. Should have written $$0 < x,y < 1$$

Will fix when I get home.

File: chapters/monte_carlo/code/julia/monte_carlo.jl [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        point_x = rand()
+        point_y = rand()
+
+        if (in_circle(point_x, point_y, radius))
+            pi_count += 1
+        end
+    end
+
+    pi_estimate = 4*pi_count/(n*radius^2)
+    println("Percent error is: ", signif(100*(pi - pi_estimate), 3), " %")
```
Comment (jiegillet) [inline chapters/monte_carlo/code/julia/monte_carlo.jl]: Here you need to divide (pi - pi_estimate) by pi

Comment (leios) [reply chapters/monte_carlo/code/julia/monte_carlo.jl]: Yup, you are right. Fixing soon. Thanks!

**Pull request merged**